### DISCUSSION TITLE
Publish Multiple Payloads through MQTT

### DISCUSSION THREAD

**Discussion by sworcester-usgs** [aws/aws-iot-device-sdk-python#341] [Q&A] [answered]
I know that I cannot send a file larger than 128KB, or it will be rejected. So I know I'll have to break up larger files and send in blocks. But I've been searching for a while, and can't seem to find anything on how I'll put the blocks "back together", in the correct order of course, after each block has been sent. Can anyone point me in the right direction for this?

**Comment by bretambrose** [comment]
AWS IoT Core has support for mqtt-based file transfers: https://docs.aws.amazon.com/iot/latest/developerguide/mqtt-based-file-delivery.html

Neither the v1 nor v2 device SDKs support this functionality.  It is generally assumed that if your device has the resources to use the device SDK then it has the resources to use http-based transfers.  The embedded C SDK, which is designed for extremely resource-constrained hardware, may have support for mqtt based file transfers, or use it in their OTA update feature.

**Reply by sworcester-usgs** [reply]
I must have posted my question in the wrong place.. I am using AWS IoT Core for my file transfers, but didn't see an aws-iot-core directory here to post my question to. Did I just miss it?

**Reply by jmklix** [reply, marked as answer]
There is no repository for AWS IoT Core, as it is not open source. This isn't the correct place to ask this question, but I can try to help you with an answer:

[This page](https://docs.aws.amazon.com/iot/latest/developerguide/mqtt-based-file-delivery-in-devices.html) describes the details of what you're looking to do. It looks like you'll want to make a `GetStream` JSON request. Then you can put the blocks back together in the order of "i" which: "is the ID of the data block contained in the payload. Data blocks are numbered starting from 0." You can use that to reassemble your file
